### PR TITLE
i12: extra debug

### PR DESCRIPTION
**Pull request by richfitz**

### CODE REVIEW

File: src/dopri.c [hunk 1]
```diff
@@ -731,7 +727,7 @@ const double* dopri_find_time(dopri_data *obj, double t) {
     const double
       t0 = ((double*) ring_buffer_tail(obj->history))[idx_t],
       t1 = ((double*) ring_buffer_tail_offset(obj->history, n - 1))[idx_t];
-    idx0 = (t1 - t0) / (n - 1);
+    idx0 = min_size((t - t0) / (t1 - t0) / (n - 1), n - 1);
```
Comment (richfitz) [inline src/dopri.c:730]: This is an unrelated bug fix that turned up debugging the malaria model - the initial interpolation to start the delay search from was completely incorrect. That was ok except it made things a bit slower unless it also out it out of range which caused a crash

File: src/dopri.c [hunk 2]
```diff
@@ -805,7 +801,56 @@ void ylag_vec_int(double t, const int *idx, size_t nidx, double *y) {
...
+}
+
+
+void dopri_callback(dopri_data *obj, double t, double h, double *y) {
+  SEXP callback = VECTOR_ELT(obj->callback, 0);
+  SEXP env = VECTOR_ELT(obj->callback, 1);
+
+  SEXP r_t = PROTECT(ScalarReal(obj->t));
+  SEXP r_h = PROTECT(ScalarReal(h));
+  SEXP r_y = PROTECT(allocVector(REALSXP, obj->n));
```
Comment (r-ash) [inline src/dopri.c:840]: Is there a reason here that `t` and `*y` are passed in but are then not used in method?

**Review by r-ash — COMMENTED** [review]
Looks good, I feel this has been super useful to review for my understanding.

Just one question.

File: src/dopri.c (hunk 2 above)
Comment (r-ash) [reply src/dopri.c:840]: Ah I thought this would show the relevant lines. The comment above refers to lines 834-845.

Comment (richfitz) [reply src/dopri.c:840]: yeah, this is an error - fixing it up now and adding a better test

**Review by r-ash — APPROVED** [review]
